regolith/tools.py

```diff
@@ -2102,6 +2105,176 @@ def google_cal_auth_flow():
+    """
+    if 'repos' in rc:
+        if rc.repos:
+            if target_repo in rc.repos:
+                if rc.repos[target_repo]:
+                    if 'params' in rc.repos[target_repo]:
+                        if rc.repos[target_repo]['params']:
+                            if 'namespace_id' in rc.repos[target_repo]['params'] and 'name' in rc.repos[target_repo]['params']:
+                                if rc.repos[target_repo]['params']['namespace_id'] and rc.repos[target_repo]['params']['name']:
+                                    if type(int(rc.repos[target_repo]['params']['namespace_id'])) != int or type(rc.repos[target_repo]['params']['name']) != str:
+                                        return False
+                                else:
+                                    return False
+                            else:
+                                return False
+                        else:
+                            return False
+                    else:
+                        return False
+                    if 'url' in rc.repos[target_repo]:
+                        if rc.repos[target_repo]['url']:
+                            url = urlparse(rc.repos[target_repo]['url'])
+                            if url[:3]:
+                                return True
+                        else:
+                            return False
+                    else:
+                        return False
+                else:
+                    return False
+            else:
+                return False
+        else:
+            return False
+    else:
+        return False
+
+
+def token_info_complete(token_info, rc):
+    """Checks if API authentication token is defined and valid in rc
+
+    Parameters:
+        token_info - string
+            the personal access token defined in rc (e.g. 'priv_token' or 'gitlab_private_token')
+            (set up in user.json)
+        rc - run control object
+
+    Returns:
+        The token if the token exists and False if not
+    """
+
+    if token_info in rc:
+        if rc.__getattr__(token_info):
+            return rc.__getattr__(token_info)
+        else:
+            return False
+    else:
+        return False
+
+def create_repo(name, target_repo, token_info, rc):
+    """ Creates a repo if repo information and token is defined in rc
+
+    Parameters:
+        name - string
+            the name of the repo being created (e.g. 'test repo')
+        target_repo - string
+            the key in rc for target repo information (e.g. 'repo1')
+        token_info - string
+            the key in rc for token (e.g. 'priv_token')
+        rc - run control object
+
+    Returns:
+        Success message (repo target_repo has been created in talks) if repo is successfully created in target_repo
+        Warning/setup messages if unsuccessful (or if repo info or token are not valid)
+    """
+    repo_info_valid = repo_info_complete(target_repo, rc)
```

Comment:
this is not true.  You have not validated that the repo is valid.  Only that something is entered for the bits and pieces.